api/src/api/user.js

```diff
@@ -0,0 +1,32 @@
+const express = require('express');
+const router = express.Router();
+const { errorWrap } = require('../middleware');
+const { sendResponse } = require('../utils/response');
+const { models } = require('../models/index.js');
+
+router.post(
+  '/',
+  errorWrap(async (req, res) => {
+    const userInfo = req.body;
```

Comment:
I believe you don't have to do userInfo.role, userInfo.authId,....

Instead, you can do `const newUser = new models.User(userInfo); `

I believe that any missing fields will be caught by Mongoose, and any additional fields won't be added to the database.

Reply:
Update on this: can you manually set the user role to ROLE_ENUM.USER? Also, adminLanguages, learnerLanguages, and collaboratorLanguages can be manually set to [].